src/components/Register/Register.js

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+import { NativeModules } from 'react-native';
+import { Text, Button, Box, FormControl, ScrollView } from 'native-base';
+import { useToggle } from '../../hooks/useToggle';
+import { Username } from './Username';
+import { KBA } from './KBA';
+import { Password } from './Password';
+import { TextField } from './TextField';
+import { Specials } from './Specials';
+import { registrationTypeFactory } from './typeFactory.js';
+
+const { ForgeRockModule } = NativeModules;
+
+const callbackTypeToComponent = {
+  ValidatedCreateUsernameCallback: ({ label, setter }) => (
+    <Username label={label} setUsername={setter} key={label} />
+  ),
+  StringAttributeInputCallback: ({ label, setter, val }) => (
+    <TextField label={label} setter={setter} key={label} val={val} />
+  ),
+  BooleanAttributeInputCallback: ({ label, setter, val }) => (
+    <Specials label={label} setter={setter} val={val} key={label} />
+  ),
+  ValidatedCreatePasswordCallback: ({ label, setter, val }) => (
+    <Password label={label} setter={setter} val={val} key={label} />
+  ),
+  TermsAndConditionsCallback: ({ label, setter, val, terms }) => (
+    <Specials
+      label={label}
+      setter={setter}
+      val={val}
+      terms={terms}
+      key={label}
+    />
+  ),
+  KbaCreateCallback: ({ label, setter, val, questions }) => (
+    <KBA
+      label={label}
+      setter={setter}
+      val={val}
+      questions={questions}
+      key={label}
+    />
+  ),
+};
+
+function RegisterContainer({ data, navigation }) {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [first, setFirst] = useState('');
+  const [last, setLast] = useState('');
+  const [email, setEmail] = useState('');
+  const [securityQuestion, setSecurityQuestion] = useState({
+    question: '',
+    answer: '',
+  });
+  const [updates, setUpdates] = useToggle(false);
+  const [specials, setSpecials] = useToggle(false);
+  const [terms, setTerms] = useToggle(false);
+
+  const getValueByType = registrationTypeFactory(
+    username,
+    password,
+    first,
+    last,
+    email,
+    specials,
+    updates,
+    securityQuestion,
+    terms,
+  );
+
+  const setStateByType = registrationTypeFactory(
+    setUsername,
+    setPassword,
+    setFirst,
+    setLast,
+    setEmail,
+    setSpecials,
+    setUpdates,
+    setSecurityQuestion,
+    setTerms,
+  );
```

Comment:
This may just be some interim code, but I'd like for us to avoid this in the final version as I'm finding it hard to wrap my brain around how it is currently being used.